File: code-coaches/hard/new-drivers-license/NewDriversLicense.swift
import Foundation

func getWaitingTime(myName: String, numAgents: Int, customers: [String]) -> Int {
    let timeForLicense = 20
    let positionInQueue = Int(ceil(Double(customers.firstIndex(of: myName)! + 1) / Double(numAgents)))
    let waitingTime = positionInQueue * timeForLicense
    return waitingTime
}

func main() {
    let myName = readLine()!
    let numAgents = Int(readLine()!)!
	let customers = readLine()!.split(separator: " ").map({ String($0) })
	
	var sortedCustomers = customers
	sortedCustomers.append(myName)
	sortedCustomers.sort()

	print(getWaitingTime(myName: myName, numAgents: numAgents, customers: sortedCustomers))
}

main()

File: code-coaches/hard/new-drivers-license/new_drivers_license.rb
def get_waiting_time(my_name, num_agents, customers)
  time_for_license = 20
  position_in_queue = (customers.index(my_name) + 1).to_f / num_agents
  waiting_time = (position_in_queue.ceil) * time_for_license
end

def main
  my_name = gets.chomp
  num_agents = gets.to_i
  customers = gets.chomp.split

  customers << my_name
  customers.sort!
  puts get_waiting_time(my_name, num_agents, customers)
end

if __FILE__ == $0
  main
end
